### PR TITLE
Add golang register info for arm64 and apple silicon

### PR DESCRIPTION
**Pull request by seekbytes**
First pull request to Ghidra, so if you have any suggestions please let me know! 

This PR adds information for registers and call conventions for golang AARCH64 and AppleSilicon (see macOS golang-based binaries).

### CODE REVIEW

File: Ghidra/Features/Base/data/noReturnFunctionConstraints.xml [hunk 1]
```diff
@@ -9,6 +9,9 @@
     <functionNamesFile>ElfFunctionsThatDoNotReturn</functionNamesFile>
   </executable_format>
   <executable_format name="Mac OS X Mach-O">
+    <compiler id="golang">
+      <functionNamesFile>GolangFunctionsThatDoNotReturn</functionNamesFile>
```
Comment (seekbytes) [inline Ghidra/Features/Base/data/noReturnFunctionConstraints.xml:13]: Is there a reason why `Mac OS X Mach-O` file format was ignored?

Comment (dev747368) [reply Ghidra/Features/Base/data/noReturnFunctionConstraints.xml:13]: mainly a detail that got overlooked.

File: Ghidra/Processors/AARCH64/data/languages/AARCH64-golang.cspec [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+				</pentry>
+                <pentry minsize="1" maxsize="8">
+					<register name="x9"/>
+				</pentry>
+                <pentry minsize="1" maxsize="8">
+					<register name="x10"/>
+				</pentry>
+                <pentry minsize="1" maxsize="8">
+					<register name="x11"/>
+				</pentry>
```
Comment (dev747368) [inline Ghidra/Processors/AARCH64/data/languages/AARCH64-golang.cspec:126]: leading whitespace should be tabs

File: Ghidra/Processors/AARCH64/data/languages/AARCH64-golang.cspec [hunk 3]
```diff
@@ -0,0 +1,202 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dev747368) [inline Ghidra/Processors/AARCH64/data/languages/AARCH64-golang.cspec:1]: there is a mixture of "-" and "_" as the separator in the filenames.  For the AARCH stuff, most of the other other files are using "_", so please rename this file to AARCH64_golang.cspec

**Comment by dev747368** [comment]
Could you:

rebase to a recent commit,
fix the leading whitespace issues
rename the cspec file
squash your commits into a single one

**Comment by cyberkaida** [comment]
@dev747368 @seekbytes I have been manually rebasing and merging this PR into my Ghidra build for a while. Would you like me to rebase, clean the white space and put up another pull request?

**Comment by seekbytes** [comment]
> I think there are still some register issues with this one too, I can fix those.

Yeah, please go ahead on opening a new PR because I didn't quite understand what was wrong...

**Comment by dev747368** [comment]
I've got some changes that should land soonish.  Take a look and see how it looks.

**Comment by cyberkaida** [comment]
OK I will hold on any work. I have an extension I was maintaining with a few extras you might want to pull in.

I extracted some additional data types that helped with function discovery etc

https://github.com/cyberkaida/golang-ghidra

**Pull request closed without merging**